pkg/plugin/processor/builtin/renameField.go

```diff
@@ -0,0 +1,149 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"context"
+	"strings"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+)
+
+type renameField struct {
+	mapping map[string]string
+
+	sdk.UnimplementedProcessor
+}
+
+func (p *renameField) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "field.rename",
+		Summary: "rename a group of fields",
+		Description: "rename a group of fields. It is not allowed to rename top-level fields (.Operation, " +
+			".Position, .Key, .Metadata, .Payload.Before, .Payload.After), processor only runs on structured data.",
+		Version: "v1.0",
```

Comment:
For the version we should probably use the full semantic version `vX.X.X`. As for the actual version to use, I'd maybe go with `v0.1.0`? Conduit itself is currently `v0.X`, it feels appropriate to keep processors also on a pre-release version. WDYT @maha-hajja @hariso?

Reply:
Agreed!